src/Microsoft.Tye.Core/DockerfileGenerator.cs

```diff
@@ -42,7 +42,7 @@ public static async Task WriteDockerfileAsync(OutputContext output, ApplicationB
             await using var stream = File.OpenWrite(filePath);
             await using var writer = new StreamWriter(stream, encoding: Encoding.UTF8, bufferSize: -1, leaveOpen: true);
 
-            var entryPoint = Path.GetFileNameWithoutExtension(project.ProjectFile.Name);
+            var entryPoint = project.AssemblyName;
```

Comment:
<3

Reply:
This is still kind a heuristic, but it's definitely a better heuristic than the project name. I think if someone where in the specialized case of using a different filename than the assembly name, we would probably say "don't do that" :)

It's a little tricky to get the actual filename that MSBuild is going to output IIRC, because it's an itemgroup and it's only created after running a bunch of targets.

Reply:
Right, it's the target path right?

Reply:
I think `TargetName` is actually what you want: https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets#L204

That's what we use to generate the name of the views assembly in razor.

Reply:
but I think it earlier SDKs it was set dynamically by CoreCompile - I'm not sure when that changed.